Automattic/woocommerce-payments
# Use client-side routing for the transaction details `ch_` → `pi_` redirect

Discovered while working on #6929.

### Changes proposed in this Pull Request

When **Transactions → Details** (`/payments/transactions/details`) is visited using a **charge ID** (`&id=ch_3Ni6ZWC15a3WbJ1t0O8hknEU`), the page must be redirected to the **payment intent ID** URL (`&id=pi_3Ni6ZWC15a3WbJ1t0WyYHnFz`). Implemented in #4348.

On `develop`, this results in a full-page reload.

This PR proposes using client-side routing to achieve this redirect, resulting in quicker page load and improved UX.

### Screenshots

Before (`develop`):

https://github.com/Automattic/woocommerce-payments/assets/3147296/b07abb47-9ed4-40a3-80a5-043657de5276


After (this PR):

https://github.com/Automattic/woocommerce-payments/assets/3147296/f65b0765-ff99-4110-890e-70d610c38576

### Testing instructions





* Navigate to Payments → Disputes and click to view the Dispute Details screen.
	* Create a test dispute by using the card `4000000000000259` at checkout
* Click the **Transaction ID: `ch_abcd`** link to view the Transaction Details screen
	<img width="337" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/f3e5d05d-bcac-42a8-8c73-9072e0e1fc0e">
* Observe the redirect that occurs on the Transaction Details screen.
	* On `develop`, the redirect will involve a full=page reload
	* On this branch, client-side routing will seamlessly handle the redirect.


-------------------


**Post merge**




#### Additional details
* In #6929, we are changing all links previously routed to **Dispute → Details** to now route to **Transactions → Details**. This is achieved by using the `dispute.charge_id` property that is readily available.
